test/lib/jdk/test/lib/artifacts/ArtifactResolver.java

```diff
@@ -68,6 +69,43 @@ public static Path resolve(String name, Map<String, Object> artifactDescription,
         return  manager.resolve(name, artifactDescription, unpack);
     }
 
+    /**
+     * Retrieve an artifact/library/file from a repository or local file system.
+     * <p>
+     * Artifacts are defined with the {@link jdk.test.lib.artifacts.Artifact}
+     * annotation. The file name should have the format ARTIFACT_NAME-VERSION.EXT
+     * <p>
+     * If you have a local version of a dependency that you want to use, you can
+     * specify that by setting the System property:
+     * <code>jdk.test.lib.artifacts.ARTIFACT_NAME</code>
+     * <p>
+     * Generally, tests that use this method should be run with <code>make test</code>.
+     * However, tests can also be run with <code>jtreg</code> but you must have a
+     * local copy of the artifact and the system property must be set as specified
+     * above.
+     *
+     * @param klass a class annotated with {@link jdk.test.lib.artifacts.Artifact}
+     * @return the local path to the artifact. If the artifact is a compressed
+     * file that gets unpacked, this path will point to the root
+     * directory of the uncompressed file.
+     * @throws IOException thrown if the artifact cannot be found
+     */
+    public static Path fetchOne(Class<?> klass) throws IOException {
+        try {
+            return ArtifactResolver.resolve(klass).entrySet().stream()
+                    .findAny().get().getValue();
+        } catch (ArtifactResolverException e) {
+            Artifact artifact = klass.getAnnotation(Artifact.class);
+            String message;
+            if (artifact != null) {
+                message = "Cannot find the artifact " + artifact.name();
+            } else {
+                message = "Class " + klass.getName() + " missing @Artifact annotation.";
```

Comment:
I assume it's a bug if `klass` is not annotated with `@Artifact`. I'd rather throw an NPE here so we can catch it.

Reply:
I updated the code to reflect that.